src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,98 @@
+package duke;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+/**
+ * Represents the file to store tasks the user has inputted.
+ */
+public class Storage {
+    private static final String DEFAULT_FILEPATH = "./data/duke.txt";
+
+    /**
+     * Creates a storage file at the intended file path.
+     */
+    public Storage() {
+    }
+
+    /**
+     * Saves user-inputted tasks into a file.
+     *
+     * @param tasks to be saved
+     * @throws IOException if file cannot be accessed.
+     */
+    public void saveTasks(ArrayList<Task> tasks) throws IOException {
+        FileWriter fw = new FileWriter(DEFAULT_FILEPATH);
+        StringBuilder textToAdd = new StringBuilder();
+        for (Task task : tasks) {
+            textToAdd.append(task).append("\n");
+        }
+        fw.write(String.valueOf(textToAdd));
+        fw.close();
+    }
+
+    /**
+     * Retrieves tasks previously saved in a file for the user.
+     * @return ArrayList of Tasks that were saved previously
+     *
+     * @throws IOException if file cannot be read.
+     * @throws DukeException if user input cannot be understood.
+     */
+    public ArrayList<Task> retrieveTasks() throws IOException, DukeException {
+        File f = new File(DEFAULT_FILEPATH);
+        f.getParentFile().mkdirs();
+        if (!f.exists()) {
+            f.createNewFile();
+        }
+        Scanner scanner = new Scanner(f);
+        ArrayList<Task> tasks = new ArrayList<>();
+        while (scanner.hasNextLine()) {
+            String line = scanner.nextLine();
+            String[] inputs = line.split("]", 3);
```

Comment:
Suggestion: Perhaps would be good to check the char at index 2 instead? so there might be less manipulation of the [].